sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java

```diff
@@ -1191,6 +1205,20 @@ public ServiceBusProcessorClientBuilder maxConcurrentCalls(int maxConcurrentCall
             return this;
         }
 
+        /**
+         * Disables auto-complete and auto-abandon of received messages. By default, a successfully processed message is
+         * {@link ServiceBusReceivedMessageContext#complete() completed}. If an error happens when
+         * the message is processed, it is {@link ServiceBusReceivedMessageContext#abandon()
+         * abandoned}.
+         *
+         * @return The modified {@link ServiceBusProcessorClientBuilder} object.
+         */
+        public ServiceBusProcessorClientBuilder disableAutoComplete() {
```

Comment:
Do you have a check that if user did not call this but  set receivemode to RECEIVE AND DELETE, that is not allowed.
We can not auto complete in that mode.

Reply:
That is for the underlying receiver to handle. If the mode is RECEIVE_AND_DELETE, `FluxAutoComplete` should not be enabled regardless of whether autocomplete is enabled or not.